### PR TITLE
ci: update docker release workflow

### PR DESCRIPTION
**Pull request by strophy**
## Description

This PR:
- Enables pushing to Docker Hub on dashevo org on release
- Enables build caching
- Disables other triggers
- Disables ARM builds (will re-enable together with other repos later)
- Tags image according to our scheme from other repos

Example build tags: https://hub.docker.com/repository/docker/strophy/tenderdash

Closes: #XXX

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -2,37 +2,15 @@ name: Build & Push
 #  Build & Push rebuilds the tendermint docker image on every push to master and creation of tags
```
Comment (QuantumExplorer) [inline .github/workflows/docker.yml:2]: rename tenderdash

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -2,37 +2,15 @@ name: Build & Push
...
 # and pushes the image to https://hub.docker.com/r/interchainio/simapp/tags
 on:
-  pull_request:
-  push:
-    branches:
-      - master
-    tags:
-      - "v[0-9]+.[0-9]+.[0-9]+" # Push events to matching v*, i.e. v1.0, v20.15.10
-      - "v[0-9]+.[0-9]+.[0-9]+-rc*" # Push events to matching v*, i.e. v1.0-rc1, v20.15.10-rc5
```
Comment (QuantumExplorer) [inline .github/workflows/docker.yml:11]: why are we removing this logic?

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -2,37 +2,15 @@ name: Build & Push
...
+  release:
+    types:
+      - published
 
 jobs:
   build:
-    runs-on: ubuntu-latest
```
Comment (QuantumExplorer) [inline .github/workflows/docker.yml:15]: Why are we not using latest?

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]

File: .github/workflows/docker.yml (hunk 2 above)
Comment (strophy) [reply .github/workflows/docker.yml:11]: To make it consistent with our other repos and how we version and release other components. If you want an image built on every push or tag that is fine as well, just let me know.

File: .github/workflows/docker.yml (hunk 3 above)
Comment (strophy) [reply .github/workflows/docker.yml:15]: So we have a stable build environment that doesn't change when GitHub feels like it, but when we feel like it. `ubuntu-latest` is current 20.04 in GitHub Actions anyway.

**Review by shumkov — APPROVED** [review]
👍

**Review by shumkov — APPROVED** [review]
LGTM! 👍

**Review by QuantumExplorer — APPROVED** [review]

**Pull request merged**